lecture_5/homework/app/controllers/buildings_controller.rb

```diff
@@ -1,7 +1,21 @@
 # frozen_string_literal: true
 
 class BuildingsController < ApplicationController
-  def index; end
+  def index
+    render json: Building.all.includes( warriors: [:weapon, :clan]), include: :warriors
```

Comment:
tego nie jestem totalnie pewien, patrzyłem w logi serwera i liczba zapytań sql spadła do 4 selectow, natomiast wydajnościowo przy takim małym zakresie danych nie było wielkich różnic. gem bullet to the rescue?